packages/journey-manager/src/index.ts

```diff
@@ -363,10 +373,14 @@ export const run = (props: RunProps): RunOutput => {
       if (action === "previous") {
         const lastTriggeredStep = triggeredSteps[triggeredSteps.length - 1];
         if (lastTriggeredStep != null) {
-          client.sendStep(lastTriggeredStep).catch((err) => {
+          client.sendStep(lastTriggeredStep.stepId).catch((err) => {
             // eslint-disable-next-line no-console
             console.warn(err);
           });
+          // Redirect to previous page if the last triggered step occurred on it
+          if (lastTriggeredStep.url !== window.location.toString()) {
+            window.location.href = lastTriggeredStep.url;
```

Comment:
Hm... 

Should we check that the trigger isn't present on the existing page (assuming the existing page also matches that triggers url filter?)

I feel like there could be a bunch of scenarios where the URL is updated with various state query parameters etc, which might cause trouble...